### PR TITLE
refactor!: combine guy_flags2 into guy_flags

### PR DESCRIPTION
**Pull request by ghost**
Squashed

### CODE REVIEW

**Comment by Jambu1989** [comment]
not possible do to how processors handle bits.

**Comment by ghost** [comment]
@connorjclark  i tried updating the replays didnt work, and the compare isnt working either.

**Comment by connorjclark** [comment]
our upload tool, surge, broke last night. https://github.com/sintaxi/surge/issues/537. I'll pin to a known working version rather than the latest

Updating replays for a refactor is very unexpected, nothing should be changing visually. That should probably be reverted and the reason for the change/regression identified.

**Comment by ghost** [comment]
traps are being weird in those 3 replays
yet they work fine everywhere else,
interestingly enough two of them are still failing which begs the question...
what is really going on here and why...

this refactor doesnt actually change behavior at all so im with connor here

File: src/zc/guys.cpp [hunk 1]
```diff
@@ -1612,7 +1612,7 @@ bool enemy::isOnSideviewPlatform()
 	int32_t usehei = (SIZEflags&guyflagOVERRIDE_HIT_HEIGHT) ? hit_height : 16;
 	if(!get_qr(qr_BROKEN_SIDEVIEW_SPRITE_JUMP)&&fall<0)
 		return false;
-	if(y + usehei >= 176 && currscr>=0x70 && !(tmpscr->flags2&wfDOWN)) return true; //Bottom of the map
+	if(y + usehei >= 176 && currscr>=0x70 && !(tmpscr->flags&wfDOWN)) return true; //Bottom of the map
```
Comment (connorjclark) [inline src/zc/guys.cpp]: revert

Comment (ghost) [reply src/zc/guys.cpp]: done

**Comment by ghost** [comment]
finally should be ready to merge now connor

**Pull request merged**